src/IntervalArith/Approx/Pi.purs

```diff
@@ -0,0 +1,137 @@
+{-
+  Based on CDAR.Approx fork by Michal Konečný (https://github.com/michalkonecny/cdar)
+  originally developed by Jens Blanck (https://github.com/jensblanck/cdar).
+-}
+module IntervalArith.Approx.Pi where
+
+import Prelude
+
+import Data.Foldable (product, sum)
+import Data.List.Lazy (List, (:))
+import Data.List.Lazy as L
+import Data.Maybe (Maybe(..))
+import Data.Tuple (Tuple(..))
+import Data.Unfoldable (unfoldr)
+import Effect.Exception.Unsafe (unsafeThrow)
+import IntervalArith.Approx (Approx, Precision, boundErrorTermMB, fromDyadicMB, fromInteger, fudge, limitAndBound, precision, recipA, setMB)
+import IntervalArith.Approx.Sqrt (sqrtA)
+import IntervalArith.Dyadic ((:^))
+import IntervalArith.Extended (Extended(..))
+import IntervalArith.Misc (Integer, big, (^))
+import Misc.LazyList (unsafeHead, unsafeTail, zipWith4, (!!))
+import Misc.LazyList.NumberSequences (ones, onetwoetc)
+import Misc.Tuple (Tuple22, tuple22)
+
+{-|
+Binary splitting summation of linearly convergent series as described in
+/'Fast multiprecision evaluation of series of rational numbers'/ by B Haible
+and T Papanikolaou, ANTS-III Proceedings of the Third International Symposium
+on Algorithmic Number Theory Pages 338-350, 1998.
+
+The main idea is to balance the computations so that more operations are
+performed with values of similar size. Using the underlying fast
+multiplication algorithms this will give performance benefits.
+
+The algorithm parallelises well. However, a final division is needed at the
+end to compute /T\/BQ/ which amount to a substantial portion of the
+computation time.
+-}
+abpq ::
```

Comment:
Could this function name be more descriptive?

Reply:
A bit like with fudge, I am reluctant to deviate from CDAR although I agree that this name does not help very much work out what the function does.  Fortunately, the comment is quite helpful in pointing us to the research paper from which this comes should we need a full explanation.